framework/src/base/Capabilities.C

```diff
@@ -0,0 +1,87 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Capabilities.h"
+#include "MooseUtils.h"
+#include "Conversion.h"
+
+#include "nlohmann/json.h"
+
+#include <vector>
+
+namespace Moose
+{
+
+Capabilities &
+Capabilities::getCapabilityRegistry()
+{
+  // We need a naked new here (_not_ a smart pointer or object instance) due to what seems like a
+  // bug in clang's static object destruction when using dynamic library loading.
+  static Capabilities * capability_registry = nullptr;
+  if (!capability_registry)
+    capability_registry = new Capabilities();
+  return *capability_registry;
+}
+
+void
+Capabilities::add(const std::string & raw_capability,
+                  CapabilityUtils::Type value,
+                  const std::string & doc)
+{
+  const auto capability = MooseUtils::toLower(raw_capability);
+  if (std::holds_alternative<std::string>(value))
+    value = MooseUtils::toLower(std::get<std::string>(value));
+
+  auto it_pair = _capability_registry.lower_bound(capability);
+  if (it_pair == _capability_registry.end() || it_pair->first != capability ||
+      it_pair->second.first == value)
+    it_pair = _capability_registry.emplace_hint(it_pair, capability, std::make_pair(value, doc));
```

Comment:
```suggestion
    _capability_registry.emplace_hint(it_pair, capability, std::make_pair(value, doc));
```

Was confused briefly as to why you used this, looks like you don't